src/OmniSharp.MSBuild/SnapSdkResolver.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using System.IO;
+
+using SdkResolverBase = Microsoft.Build.Framework.SdkResolver;
+using SdkResolverContextBase = Microsoft.Build.Framework.SdkResolverContext;
+using SdkResultBase = Microsoft.Build.Framework.SdkResult;
+using SdkResultFactoryBase = Microsoft.Build.Framework.SdkResultFactory;
+
+namespace Microsoft.Build.BackEnd.SdkResolution
+{
+    /// <summary>
+    ///     Resolve SDKs that are part of a Snap installation
+    /// <remarks>
+    ///     This will search for the Snap SDK installation of the .NET Core SDK
+    ///     by executing `dotnet --list-sdks`.
+    /// </remarks>
+    /// </summary>
+    internal class SnapSdkResolver : SdkResolverBase
```

Comment:
Is this picked up just passively by MSBuild (by inheriting from `Microsoft.Build.Framework.SdkResolver`), or does it require additional registration in MEF?